src/core/runtime.d

```diff
@@ -328,46 +354,58 @@ struct Runtime
      *
      *             if (fp)
      *             {
+     *                 ++nRun;
      *                 try
      *                 {
      *                     fp();
      *                 }
      *                 catch (Throwable e)
      *                 {
      *                     writeln(e);
-     *                     failed++;
+     *                     --failed;
      *                 }
      *             }
      *         }
      *     }
-     *     return failed == 0;
+     *     return failed == 0 ? nRun : failed;
      * }
      * ---------
      */
     static @property void moduleUnitTester( ModuleUnitTester h )
     {
         sm_moduleUnitTester = h;
+        sm_boolModuleUnitTester = null;
     }
 
+    /// Ditto
+    static @property void moduleUnitTester( BoolModuleUnitTester h )
+    {
+        sm_moduleUnitTester = null;
+        sm_boolModuleUnitTester = h;
+    }
 
     /**
-     * Gets the current module unit tester.
+     * Gets the current module unit tester. Note that if a boolean returning
+     * module unit tester is set, it will be returned, even though the type is
+     * not exactly the same.
      *
      * Returns:
      *  The current module unit tester handler or null if none has been set.
      */
     static @property ModuleUnitTester moduleUnitTester()
     {
+        if(sm_moduleUnitTester == null)
+            return cast(ModuleUnitTester)sm_boolModuleUnitTester;
```

Comment:
Also note, that there isn't any danger of improperly calling the function, since it takes no parameters. Just the return type is different, and won't crash the code that uses the wrong function type.